service/history/decision/handler_test.go

```diff
@@ -462,11 +472,43 @@ func (s *DecisionHandlerSuite) TestHandleDecisionTaskStarted() {
 			expectErr: nil,
 			mutablestate: &persistence.WorkflowMutableState{
 				ExecutionInfo: &persistence.WorkflowExecutionInfo{
-					DecisionScheduleID: 0,
-					NextEventID:        3,
-					DecisionRequestID:  "test-request-id",
+					NextEventID:       3,
+					DecisionRequestID: "test-request-id",
+					DecisionAttempt:   1,
 				},
 			},
+			assertCalls: func(resp *types.RecordDecisionTaskStartedResponse) {
+				// expect test.mutablestate.ExecutionInfo.DecisionAttempt
+				s.Equal(int64(1), resp.DecisionInfo.ScheduledEvent.DecisionTaskScheduledEventAttributes.Attempt)
+			},
+		},
+		{
+			name:     "success - decision startedID is empty",
+			domainID: _testDomainUUID,
+			expectCalls: func(h *handlerImpl) {
+				h.shard.(*shard.MockContext).EXPECT().GetEventsCache().Times(1).Return(events.NewMockCache(s.controller))
+				h.shard.(*shard.MockContext).EXPECT().GenerateTransferTaskIDs(gomock.Any()).Times(1).Return([]int64{0}, nil)
+				h.shard.(*shard.MockContext).EXPECT().
+					AppendHistoryV2Events(gomock.Any(), gomock.Any(), _testDomainUUID, types.WorkflowExecution{WorkflowID: _testWorkflowID, RunID: _testRunID}).
+					Return(&persistence.AppendHistoryNodesResponse{}, nil)
+				h.shard.(*shard.MockContext).EXPECT().UpdateWorkflowExecution(gomock.Any(), gomock.Any()).Return(&persistence.UpdateWorkflowExecutionResponse{MutableStateUpdateSessionStats: &persistence.MutableStateUpdateSessionStats{}}, nil)
+				h.shard.(*shard.MockContext).EXPECT().GetShardID().Return(_testShardID)
+				engine := engine.NewMockEngine(s.controller)
+				h.shard.(*shard.MockContext).EXPECT().GetEngine().Times(3).Return(engine)
+				engine.EXPECT().NotifyNewHistoryEvent(gomock.Any())
+				engine.EXPECT().NotifyNewTransferTasks(gomock.Any())
+				engine.EXPECT().NotifyNewTimerTasks(gomock.Any())
+				engine.EXPECT().NotifyNewCrossClusterTasks(gomock.Any())
+				engine.EXPECT().NotifyNewReplicationTasks(gomock.Any())
+			},
+			expectErr: nil,
+			mutablestate: &persistence.WorkflowMutableState{
+				ExecutionInfo: &persistence.WorkflowExecutionInfo{
+					DecisionStartedID: -23,
+					NextEventID:       2,
+				},
+			},
+			assertCalls: func(resp *types.RecordDecisionTaskStartedResponse) {},
```

Comment:
Should we assert the response body like the previous success?

Reply:
Yes, thank you! Added in #5932